common/lib/xmodule/xmodule/js/src/capa/display.js

```diff
@@ -0,0 +1,1319 @@
+                            break;
+                        default:
+                            that.gentle_alert(response.success);
+                        }
+                        return Logger.log('problem_graded', [that.answers, response.contents], that.id);
+                    }
+                };
+                $.ajaxWithPrefix('' + this.url + '/problem_check', settings);
+            }
+        };
+
+        Problem.prototype.submit = function() {
+            if (!this.submit_save_waitfor(this.submit_internal)) {
+                this.disableAllButtonsWhileRunning(this.submit_internal, true);
+            }
+        };
+
+        Problem.prototype.submit_internal = function() {
+            var that = this;
+            Logger.log('problem_check', this.answers);
+            return $.postWithPrefix('' + this.url + '/problem_check', this.answers, function(response) {
+                switch (response.success) {
+                case 'incorrect':
+                case 'correct':
+                    window.SR.readTexts(that.get_sr_status(response.contents));
+                    that.el.trigger('contentChanged', [that.id, response.contents]);
+                    that.render(response.contents, that.focus_on_submit_notification);
+                    that.updateProgress(response);
+                    break;
+                default:
+                    that.saveNotification.hide();
+                    that.gentle_alert(response.success);
+                }
+                return Logger.log('problem_graded', [that.answers, response.contents], that.id);
+            });
+        };
+
+        Problem.prototype.get_sr_status = function(contents) {
+            /**
+             * This method builds up an array of strings to send to the page screen-reader span.
+             * It first gets all elements with class "status", and then looks to see if they are contained
+             * in sections with aria-labels. If so, labels are prepended to the status element text.
+             * If not, just the text of the status elements are returned.
+             */
+            var addedStatus, ariaLabel, element, labeledStatus, parentSection, statusElement, template, i, len;
+            statusElement = $(contents).find('.status');
+            labeledStatus = [];
+            for (i = 0, len = statusElement.length; i < len; i++) {
+                element = statusElement[i];
+                parentSection = $(element).closest('section');
+                addedStatus = false;
+                if (parentSection) {
+                    ariaLabel = parentSection.attr('aria-label');
+                    if (ariaLabel) {
+                        // Translators: This is only translated to allow for reordering of label and associated status.;
+                        template = gettext('{label}: {status}');
+                        labeledStatus.push(edx.StringUtils.interpolate(template, {
```

Comment:
Unless you think that the variable "template" will be removed and the content in-lined before the extraction?